src/components/search/BreadCrumbs.js

```diff
@@ -2,47 +2,126 @@ import React from 'react';
+                                    <RiCloseFill onClick={() => handleRemoveFacet(facet, value)} />
+                                </Button>&nbsp;&nbsp;
+                            </span>
+                        )) : (
+                            <span key={facet + "_breadcrumb"}>
+                                <Button variant="outline-secondary">
+                                    {`${facet.replace('datePubmedAdded', 'Date Added to PubMed')
+                                        .replace('datePubmedModified', 'Date Modified in PubMed')
+                                        .replace('datePublished', 'Date Published')
+                                        .replace('dateCreated', 'Date Created')}`} &nbsp;
+                                    <RiCloseFill onClick={() => handleRemoveDate(facet)} />
+                                </Button>&nbsp;&nbsp;
+                            </span>
+                        )
+                    ))}
+                    {Object.entries(searchExcludedFacetsValues).map(([facet, values]) =>
+                        values.map(value =>
+                            <span key={facet + value + "_breadcrumb"}>
+                                <Button variant="outline-danger">Exclude&nbsp;{(RENAME_FACETS.hasOwnProperty(facet) ? RENAME_FACETS[facet] : facet.replace('.keyword', '').replaceAll('_', ' ')) + ": " + value} &nbsp;
+                                    <RiCloseFill onClick={() => dispatch(removeExcludedFacetValue(facet, value))} />
+                                </Button>&nbsp;&nbsp;
+                            </span>
+                        )
+                    )}
+                    {datePubmedAdded && (
+                        <span key="datePubmedAdded_breadcrumb">
+                            <Button variant="outline-secondary">
+                                Date Added to PubMed &nbsp;
+                                <RiCloseFill onClick={() => handleRemoveDate('datePubmedAdded')} />
+                            </Button>&nbsp;&nbsp;
                         </span>
                     )}
-                    {Object.entries(searchExcludedFacetsValues).map(([facet, values]) =>
-                        <span key={facet + "_group"}>
-                            {values.map(value =>
-                                <span key={value + "_breadcrumb"}>
-                                    <Button variant="outline-danger">Exclude&nbsp;{(RENAME_FACETS.hasOwnProperty(facet) ? RENAME_FACETS[facet] : facet.replace('.keyword', '').replaceAll('_', ' ')) + ": " + value} &nbsp;
-                                        <RiCloseFill onClick={() => dispatch(removeExcludedFacetValue(facet, value))}/>
-                                    </Button>&nbsp;&nbsp;
-                                </span>)}
+                    {datePubmedModified && (
```

Comment:
@sweng66 can you refactor this part and create a separate component to avoid repeating the same code multiple times?

Reply:
Will try to do so. Thanks !! @valearna

Reply:
fixed